package-builder/build-manifest.nix

```diff
@@ -162,10 +153,12 @@ pkgs.runCommandNoCC name
               ''
           }
 
-          # Run the build script using _BOTH_ the flox and build wrapper
-          # environments, ensuring that the build wrapper environment is the
-          # "inner" activation so that its tools and libraries are preferred
-          # over those from the "develop" environment.
+          # Perform the build using a _copy_ of the "develop" environment
+          # found in a storepath with the same strlen as the "build-wrapper"
+          # environment, so that following a successful build we can replace
+          # references to the former with the latter as we copy the output
+          # to its final path.
+
```

Comment:
I think this still needs a sentence as to _why_ we do this, i.e. reduce the size of the closure (and maybe reduce circular references with older selves? not sure about that one).
I think having those comments above might be enough though.

Reply:
I think it's understood that the reader of this file would understand the reasons for the runtime-packages feature. Agree that this comment in this part of the file is more about the implementation than the reason.